core/kotlinx-coroutines-test/src/TestDispatchers.kt

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2016-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
+ */
+@file:Suppress("unused")
+@file:JvmName("TestDispatchers")
+
+package kotlinx.coroutines.test
+
+import kotlinx.coroutines.*
+import kotlinx.coroutines.test.internal.*
+
+/**
+ * Sets the given [dispatcher] as an underlying dispatcher of [Dispatchers.Main].
+ * All consecutive usages of [Dispatchers.Main] will use given [dispatcher] under the hood.
+ *
+ * It is unsafe to call this method if alive coroutines launched in [Dispatchers.Main] exist.
+ */
+public fun Dispatchers.setMain(dispatcher: CoroutineDispatcher = Dispatchers.Unconfined) {
```

Comment:
Should be marked as experimental together with `resetMain`. I would also remove default. It will make it easier to analyze what kind of dispatcher users actually need.